mlflow/projects/databricks.py

```diff
@@ -30,19 +29,154 @@
+            'new_cluster': cluster_spec,
+            'shell_command_task': {
+                'command': command,
+                "env_vars": env_vars
+            },
+            # NB: We use <= on the version specifier to allow running projects on pre-release
+            # versions, where we will select the most up-to-date mlflow version available.
+            # Also note, that we escape this so '<' is not treated as a shell pipe.
+            "libraries": [{"pypi": {"package": "'mlflow<=%s'" % VERSION}}],
+        }
+        run_submit_res = self._jobs_runs_submit(req_body_json)
+        databricks_run_id = run_submit_res["run_id"]
+        return databricks_run_id
+
+    def _before_run_validations(self, tracking_uri, cluster_spec):
+        """Validations to perform before running a project on Databricks."""
+        self._check_auth_available()
+        if cluster_spec is None:
+            raise ExecutionException("Cluster spec must be provided when launching MLflow project "
+                                     "runs on Databricks.")
+        if tracking.utils._is_local_uri(tracking_uri):
+            raise ExecutionException(
+                "When running on Databricks, the MLflow tracking URI must be set to a remote URI "
+                "accessible to both the current client and code running on Databricks. Got local "
+                "tracking URI %s." % tracking_uri)
+
+    def run_databricks(self, uri, entry_point, work_dir, parameters, experiment_id, cluster_spec,
+                       run_id):
+        tracking_uri = _get_tracking_uri_for_run()
+        self._before_run_validations(tracking_uri, cluster_spec)
+        dbfs_fuse_uri = self._upload_project_to_dbfs(work_dir, experiment_id)
+        env_vars = {
+            tracking._TRACKING_URI_ENV_VAR: tracking_uri,
+            tracking._EXPERIMENT_ID_ENV_VAR: experiment_id,
+        }
+        eprint("=== Running entry point %s of project %s on Databricks ===" % (entry_point, uri))
+        # Launch run on Databricks
+        with open(cluster_spec, 'r') as handle:
+            try:
+                cluster_spec = json.load(handle)
+            except ValueError:
+                eprint("Error when attempting to load and parse JSON cluster spec from file "
+                       "%s. " % cluster_spec)
+                raise
+        command = _get_databricks_run_cmd(dbfs_fuse_uri, run_id, entry_point, parameters)
+        return self._run_shell_command_job(uri, command, env_vars, cluster_spec)
 
 
-def _jobs_runs_submit(req_body_json):
-    return rest_utils.databricks_api_request(
-        endpoint="jobs/runs/submit", method="POST", json=req_body_json)
+def _get_tracking_uri_for_run():
+    if not tracking.utils.is_tracking_uri_set():
+        return "databricks"
+    uri = tracking.get_tracking_uri()
+    if uri.startswith("databricks"):
+        return "databricks"
+    return uri
 
 
 def _get_databricks_run_cmd(dbfs_fuse_tar_uri, run_id, entry_point, parameters):
```

Comment:
No changes to this guy, right?

Reply:
Yep, no need to change this to account for profile information